### PR TITLE
FINERACT-1724: Sonar fixes

### PR DESCRIPTION
**Pull request by abraham-menyhart**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [ ] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [ ] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-client/src/main/java/org/apache/fineract/client/util/CallFailedRuntimeException.java [hunk 1]
```diff
@@ -6,9 +6,9 @@
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License. You may obtain a copy of the License at
- *
+ * <p>
```
Comment (adamsaghy) [inline fineract-client/src/main/java/org/apache/fineract/client/util/CallFailedRuntimeException.java]: Please undo this changes!

File: fineract-client/src/test/java/org/apache/fineract/client/test/FineractClientTechnicalTest.java [hunk 2]
```diff
@@ -41,8 +41,7 @@ void testInvalidOperations() {
 
     @Test
     void testFineractClientBuilder() {
-        assertThrows(IllegalStateException.class, () -> {
-            FineractClient.builder().build();
-        });
+        FineractClient.Builder fineractClientBuilder = FineractClient.builder();
```
Comment (adamsaghy) [inline fineract-client/src/test/java/org/apache/fineract/client/test/FineractClientTechnicalTest.java]: This refactor looks useless to me. What was the idea behind the change?

File: fineract-core/src/main/java/org/apache/fineract/infrastructure/core/data/DataValidatorBuilder.java [hunk 3]
```diff
@@ -1032,7 +1009,7 @@ public DataValidatorBuilder validatePhoneNumber() {
          */
         final String regex = "^\\+?[0-9. ()-]{0,25}$";
         final Pattern pattern = Pattern.compile(regex);
-        final Matcher matcher = pattern.matcher(this.value.toString());
+        final Matcher matcher = pattern.matcher(this.value != null ? this.value.toString() : "");
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/infrastructure/core/data/DataValidatorBuilder.java]: this extra condition is useless, the null check was done a couple lines above. Also even if we are remove that check, using empty string might not be the wanted behaviour!

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 4]
```diff
@@ -1051,8 +1051,8 @@ private void parseDatatableColumnForDrop(final JsonObject column, StringBuilder
                 .append(" WHERE i.CONSTRAINT_TYPE = 'FOREIGN KEY'").append(" AND i.TABLE_SCHEMA = DATABASE()")
                 .append(" AND i.TABLE_NAME = '").append(datatableName).append("' AND i.CONSTRAINT_NAME = 'fk_").append(datatableAlias)
                 .append("_").append(name).append("' ");
-        final int count = this.jdbcTemplate.queryForObject(findFKSql.toString(), Integer.class);
-        if (count > 0) {
+        final Integer count = this.jdbcTemplate.queryForObject(findFKSql.toString(), Integer.class);
+        if (count != null && count > 0) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: COUNT function never returns null

File: fineract-core/src/main/java/org/apache/fineract/accounting/glaccount/service/GLAccountWritePlatformServiceJpaRepositoryImpl.java [hunk 5]
```diff
@@ -170,8 +170,8 @@ public CommandProcessingResult updateGLAccount(final Long glAccountId, final Jso
     private void validateForAttachedProduct(Long glAccountId) {
         String sql = "select count(*) from acc_product_mapping acc where acc.gl_account_id = ?";
         try {
-            int count = this.jdbcTemplate.queryForObject(sql, Integer.class, glAccountId);
```
Comment (adamsaghy) [inline fineract-core/src/main/java/org/apache/fineract/accounting/glaccount/service/GLAccountWritePlatformServiceJpaRepositoryImpl.java]: COUNT function never returns null.

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 6]
```diff
@@ -1868,7 +1868,7 @@ private String queryForApplicationTableName(final String datatable) {
     private void validateDatatable(final String datatable) {
         final String sql = "SELECT COUNT(application_table_name) FROM x_registered_table WHERE registered_table_name = ?";
         final Integer count = jdbcTemplate.queryForObject(sql, Integer.class, datatable);
-        if (count == 0) {
+        if (count != null && count == 0) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java [hunk 7]
```diff
@@ -2278,7 +2278,7 @@ public boolean isDatatableAttachedToEntityDatatableCheck(final String datatableN
         builder.append(" JOIN m_entity_datatable_check edc ON edc.x_registered_table_name = xrt.registered_table_name");
         builder.append(" WHERE edc.x_registered_table_name = '" + datatableName + "'");
         final Long count = this.jdbcTemplate.queryForObject(builder.toString(), Long.class);
-        return count > 0 ? true : false;
+        return count != null && count > 0;
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/SchedulerJobRunnerReadServiceImpl.java [hunk 8]
```diff
@@ -137,8 +137,8 @@ private boolean isJobExist(final Long jobId) {
         boolean isJobPresent = false;
         try {
             final String sql = "select count(*) from job job where job.id= ?";
-            final int count = this.jdbcTemplate.queryForObject(sql, Integer.class, new Object[] { jobId });
-            if (count == 1) {
+            final Integer count = this.jdbcTemplate.queryForObject(sql, Integer.class, new Object[] { jobId });
+            if (count != null && count == 1) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/jobs/service/SchedulerJobRunnerReadServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/service/TwoFactorConfigurationServiceImpl.java [hunk 9]
```diff
@@ -133,7 +133,7 @@ public boolean isSMSEnabled() {
     @Cacheable(value = "tfConfig", key = "T(org.apache.fineract.infrastructure.core.service.ThreadLocalContextUtil).getTenant().getTenantIdentifier()+'|smsProvider'")
     public Integer getSMSProviderId() {
         Integer value = getIntegerConfig(TwoFactorConfigurationConstants.SMS_PROVIDER_ID, null);
-        if (value < 1) {
+        if (value != null && value < 1) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/service/TwoFactorConfigurationServiceImpl.java]: This looks wrong to me. Should it be:
IF value is NULL OR value < 1 THEN return null?

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/service/CalendarReadPlatformServiceImpl.java [hunk 10]
```diff
@@ -264,8 +264,8 @@ public Boolean isCalendarAssociatedWithEntity(final Long entityId, final Long ca
         String query = "Select COUNT(*) from m_calendar_instance ci where ci.entity_id = ? and ci.calendar_id = ? and "
                 + " ci.entity_type_enum = ?";
         try {
-            int calendarInstaneId = this.jdbcTemplate.queryForObject(query, Integer.class, entityId, calendarId, entityTypeId);
-            if (calendarInstaneId > 0) {
+            Integer calendarInstaneId = this.jdbcTemplate.queryForObject(query, Integer.class, entityId, calendarId, entityTypeId);
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/calendar/service/CalendarReadPlatformServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanArrearsAgingServiceImpl.java [hunk 11]
```diff
@@ -99,7 +99,7 @@ public void registerForNotification() {
 
     @Override
     public void updateLoanArrearsAgeingDetailsWithOriginalSchedule(final Loan loan) {
-        int count = this.jdbcTemplate.queryForObject("select count(mla.loan_id) from m_loan_arrears_aging mla where mla.loan_id =?",
+        Integer count = this.jdbcTemplate.queryForObject("select count(mla.loan_id) from m_loan_arrears_aging mla where mla.loan_id =?",
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanArrearsAgingServiceImpl.java:102]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanArrearsAgingServiceImpl.java [hunk 12]
```diff
@@ -108,7 +108,7 @@ public void updateLoanArrearsAgeingDetailsWithOriginalSchedule(final Loan loan)
         if (scheduleDate.size() > 0) {
             List<Map<String, Object>> transactions = getLoanSummary(loan.getId(), loan.getLoanSummary());
             updateScheduleWithPaidDetail(scheduleDate, transactions);
-            createInsertStatements(updateStatement, scheduleDate, count == 0);
+            createInsertStatements(updateStatement, scheduleDate, (count != null && count == 0));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanArrearsAgingServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanArrearsAgingServiceImpl.java [hunk 13]
```diff
@@ -126,9 +126,9 @@ public Map<Long, List<LoanSchedulePeriodData>> getScheduleDate(String loanId) {
 
     @Override
     public void updateLoanArrearsAgeingDetails(final Loan loan) {
-        int count = this.jdbcTemplate.queryForObject("select count(mla.loan_id) from m_loan_arrears_aging mla where mla.loan_id =?",
+        Integer count = this.jdbcTemplate.queryForObject("select count(mla.loan_id) from m_loan_arrears_aging mla where mla.loan_id =?",
                 Integer.class, loan.getId());
-        String updateStatement = constructUpdateStatement(loan, count == 0);
+        String updateStatement = constructUpdateStatement(loan, (count != null && count == 0));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanArrearsAgingServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/registration/service/SelfServiceRegistrationReadPlatformServiceImpl.java [hunk 14]
```diff
@@ -42,7 +42,7 @@ public boolean isClientExist(String accountNumber, String firstName, String last
             params = new Object[] { accountNumber, firstName, lastName, mobileNumber };
         }
         Integer count = this.jdbcTemplate.queryForObject(sql, Integer.class, params);
-        if (count == 0) {
+        if (count != null && count == 0) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/self/registration/service/SelfServiceRegistrationReadPlatformServiceImpl.java]: COUNT function never returns null

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/PocketAccountMappingReadPlatformServiceImpl.java [hunk 15]
```diff
@@ -78,7 +80,7 @@ public PocketAccountMappingData retrieveAll() {
     public boolean validatePocketAndAccountMapping(Long pocketId, Long accountId, Integer accountType) {
         final String sql = "select count(id) from m_pocket_accounts_mapping mapping where pocket_id = ? and account_id = ? and account_type = ?";
         try {
-            return this.jdbcTemplate.queryForObject(sql, Boolean.class, pocketId, accountId, accountType);
+            return TRUE.equals(this.jdbcTemplate.queryForObject(sql, Boolean.class, pocketId, accountId, accountType));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/self/pockets/service/PocketAccountMappingReadPlatformServiceImpl.java]: Not the best solution:
Since it is executing a select with count, there will be always a result. THe EmptyResultDataAccessException will never be thrown.
Also mapping the count result to boolean is questionable. If we are touching this, we should make it better!

File: fineract-provider/src/main/java/org/apache/fineract/useradministration/service/AppUserReadPlatformServiceImpl.java [hunk 16]
```diff
@@ -219,7 +219,7 @@ public boolean isUsernameExist(String username) {
         String sql = "select count(*) from m_appuser where username = ?";
         Object[] params = new Object[] { username };
         Integer count = this.jdbcTemplate.queryForObject(sql, Integer.class, params);
-        if (count == 0) {
+        if (count != null && count == 0) {
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/useradministration/service/AppUserReadPlatformServiceImpl.java]: COUNT function never returns null

**Review by adamsaghy — CHANGES_REQUESTED** [review]
Please see my comments!

**Review by adamsaghy — APPROVED** [review]
LGTM

**Review by galovics — APPROVED** [review]
